pkg/controller/elasticsearch/validation/warnings.go

```diff
@@ -13,9 +13,20 @@ import (
 
 var warnings = []validation{
 	noUnsupportedSettings,
+	validClientAuthentication,
 }
 
+type nodeSetChecker func(*common.CanonicalConfig, int) field.ErrorList
+
 func noUnsupportedSettings(es esv1.Elasticsearch) field.ErrorList {
+	return checkNodeSets(es, validateSettings)
+}
+
+func validClientAuthentication(es esv1.Elasticsearch) field.ErrorList {
+	return checkNodeSets(es, validateClientAuthentication)
+}
+
+func checkNodeSets(es esv1.Elasticsearch, check nodeSetChecker) field.ErrorList {
```

Comment:
My attempt to reduce doing the same iteration in both validations. Let me know if there's a better way.

Reply:
Better is pretty relative here :) Your approach is good. I think I would just have added the new validation in the existing `noUnsupportedSettings` func.

Maybe create two functions:
```go
func noUnsupportedSettings(es esv1.Elasticsearch) field.ErrorList {
	...
		errs = append(errs, validUnsupportedSettings(config)...)
		errs = append(errs, validPartiallyUnsupportedSettings(config)...)
	}
	return errs
}

func validUnsupportedSettings(config *common.CanonicalConfig) field.ErrorList
func validPartiallyUnsupportedSettings(config *common.CanonicalConfig) field.ErrorList
```

Reply:
Makes sense, thanks for the suggestion. I made the changes, I think it's more readable now.